fix(tsp-obs): the OBS utility in GWE not supporting TEMPERATURE keyword

emorway-usgs opened the pull request:
The mf6io guide appropriately included a section on using the OBS utility with a GWE model; however, support in the MF6 source code was lacking for a TEMPERATURE `obstype`.  This issue was originally pointed out in #1944 

- [x] Replaced section above with description of pull request
- [x] Fixes issue #1944
- [x] Added new test or modified an existing test
- [x] Ran `ruff` on new and modified python scripts in .doc, autotests, doc, distribution, pymake, and utils subdirectories.
- [x] Formatted new and modified Fortran source files with `fprettify`
- [x] Updated [develop.tex](/MODFLOW-USGS/modflow6/doc/ReleaseNotes/develop.tex) with a plain-language description of the bug fix, change, feature; required for changes that may affect users

langevin-usgs reviewed (commented):
I see that `TEMPERATURE` is added as an observation type to tsp-obs, but I wonder if we need a way for the concrete models (GWE, GWT) to add these through a subclass.  FLOWJA can stay in the parent because it is available to both model types.  Otherwise, what happens when someone specifies CONCENTRATION for a GWE model or TEMPERATURE for a GWT model?  Okay with kicking this down the road for now, but this might be a good opportunity to think this through.

emorway-usgs commented:
> I see that `TEMPERATURE` is added as an observation type to tsp-obs, but I wonder if we need a way for the concrete models (GWE, GWT) to add these through a subclass. FLOWJA can stay in the parent because it is available to both model types. Otherwise, what happens when someone specifies CONCENTRATION for a GWE model or TEMPERATURE for a GWT model? Okay with kicking this down the road for now, but this might be a good opportunity to think this through.

Thanks for the nudge to do this better @langevin-usgs.  I'm hopeful that [aae590c](https://github.com/MODFLOW-USGS/modflow6/pull/1950/commits/aae590caec16cc704deb7135c58b67e6c879a9ef) addresses these concerns.  Initially I had worked up some error checking in response to this comment (which I'm pasting here in case I need to come back to it):
```
  !> @brief Check observation type against name of dependent variable
  !<
  subroutine check_obs_id(this)
    ! -- modules
    use InputOutputModule, only: upcase
    ! -- dummy
    class(TspObsType), intent(inout) :: this
    ! -- local
    integer(I4B) :: i
    character(len=LENVARNAME) :: obs_str
    character(len=LINELENGTH) :: ermsg
    character(len=*), parameter :: fmterrmsg = &
      &"(4x, 'Obs type:',x,a,x,'does not match model type:',x,a)" 
    !
    ! -- Ensure that obs type matches model type
    do i = 1, this%npakobs
      obs_str = trim(adjustl(this%pakobs(i)%obsrv%obstypeid))
      call upcase(obs_str)
      if (obs_str /= 'FLOW-JA-FACE') then
        if (obs_str /= this%depvartype) then
          ! -- Report mismatch in observation types
          write(ermsg, fmterrmsg) obs_str, &
                                  trim(adjustl(this%filtyp))
          call store_error(ermsg)
        end if
      end if
    end do
    !
    ! -- Return
    return
  end subroutine check_obs_id
```
However, I think the current offering is a bit sleeker with less code-bloat.  I discovered there is already some error checking built in which the slightly-expanded new autotest (`test_gwe_obs.py`) now leverages.  To make use of the existing error checking, the call to StoreObsType (line 100-ish in `tsp-obs.f90`) simply needed the _dynamic_ dependent variable name passed in rather than  a hard-coded name (`concentration` or `temperature`)
